packages/circuit-ui/components/RadioButton/RadioButton.tsx

```diff
@@ -33,15 +33,10 @@ import { useClickEvent, TrackingProps } from '../../hooks/useClickEvent';
 
 export interface RadioButtonProps
   extends InputHTMLAttributes<HTMLInputElement> {
-  /**
-   * @deprecated
-   * Use the label prop instead.
-   */
-  children?: ReactNode;
   /**
    * A clear and concise description of the option's purpose.
    */
-  label?: ReactNode;
+  label: ReactNode;
```

Comment:
This is also a breaking change: previously we had an optional `children` and an optional `label`.

This should be required since the accessible label is an accessibility requirement.

I'm also tempted to migrate it to `string` but I'm not sure why it was a `ReactNode` in the first place, and whether this would cause issues. But from an a11y standpoint, the label is the input's accessible name, which means that it should be as straightforward as possible—hence if I could I'd make this a string.

Reply:
Added the logic to throw an error in [421a04d](https://github.com/sumup-oss/circuit-ui/pull/1512/commits/421a04debb4cf639ad261a4b73a700ec9bc046dc), plus a new major changeset since this is an unrelated change.

Won't touch the type of the `label` prop for now.